libraries/crypto/src/cbc.rs

```diff
@@ -13,24 +13,21 @@
 // limitations under the License.
 
 use super::util::{xor_block_16, Block16};
-use super::{Decrypt16BytesBlock, Encrypt16BytesBlock};
+use crate::aes256::{DecryptionKey, EncryptionKey};
+use core::convert::TryInto;
 
-pub fn cbc_encrypt<K>(key: &K, mut iv: Block16, blocks: &mut [Block16])
-where
-    K: Encrypt16BytesBlock,
-{
-    for block in blocks {
+pub fn cbc_encrypt(key: &EncryptionKey, mut iv: Block16, blocks: &mut [u8]) {
+    for block in blocks.chunks_mut(16) {
+        let block: &mut Block16 = block.try_into().unwrap();
```

Comment:
It's a bit unfortunate that the new API potentially adds runtime checks rather than checking lengths at compile time. At least Rust nightly now has `slice::array_chunks` which would be useful here (https://doc.rust-lang.org/std/primitive.slice.html#method.array_chunks).

Reply:
The length was already checked at runtime before, but it was done in the `crypto_wrapper.rs`. Now the check is moved here and made a panic to preserve the signature. Could be made an error if needed.

Yes, there's a comment somewhere about `array_chunks` but it's still unstable.